pallets/communities/src/types.rs

```diff
@@ -0,0 +1,65 @@
+use crate::{Config, DomainNameOf};
+use frame_support::RuntimeDebug;
+use parity_scale_codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+use sp_runtime::traits::Zero;
+use sp_std::{fmt, num::ParseIntError, str::FromStr, vec::Vec};
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, MaxEncodedLen, RuntimeDebug, TypeInfo)]
+#[scale_info(skip_type_params(T))]
+#[codec(mel_bound(T: Config))]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct Community<T: Config> {
+	// TODO : Maybe every community can configure an asset?
```

Comment:
Communities indeed will have an asset, we can actually provide one automatically using the community id as asset id. Our asset type can have a variant that is a community Id(`VirtoAsset::Community(CommunityId)`). To smooth its usage we would implement `From<CommunityId> for VirtoAsset`